fireant/queries/builder.py

```diff
@@ -297,16 +301,49 @@ def sql(self):
         Serializes this query builder as a set of SQL queries.  This method will always return a list of one query since
         only one query is required to retrieve dimension choices.
 
-        This function only handles dimensions (select+group by) and filtering (where/having), which is everything needed
-        for the query to fetch choices for dimensions.
-
         The slicer query extends this with metrics, references, and totals.
         """
+
+        def in_hint_table(dim):
+            return all(dim_field.name in hint_column_names for dim_field in dim.definition.fields())
+
+        hint_column_definitions = self.dataset.database.get_column_definitions(
+              self.dataset.hint_table._schema._name,
+              self.dataset.hint_table._table_name
+        ) if self.dataset.hint_table else []
+
+        hint_column_names = [hint_column_definition[0] for hint_column_definition in hint_column_definitions]
+
+        dimensions = self._dimensions
+
+        # All definitions of all dimensions have to exist in hint table
+        is_hint_dimension = all(in_hint_table(dim) for dim in self._dimensions)
```

Comment:
i think we can assume that the definitions for all dimensions will exist in the fact table. If not, we can execute the query anyway which will lead to a SQL error that the end user will see.